pkgs/development/tools/wasmedge/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  llvm_12,
+  lld,
+  boost,
+  cmake,
+  gtest,
+  spdlog,
+}:
+stdenv.mkDerivation rec {
+  pname = "wasmedge";
+  version = "0.10.1";
+
+  src = fetchFromGitHub {
+    owner = "WasmEdge";
+    repo = "WasmEdge";
+    rev = version;
+    sha256 = "sha256-SJi8CV0sa+g+Ngvdw8+SxV3kHqoiKBhYUwDLZM4+jX0=";
+  };
+
+  buildInputs = [
+    boost
+    llvm_12
+  ];
+
+  nativeBuildInputs = [cmake lld boost gtest spdlog];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DWASMEDGE_BUILD_TESTS=OFF" # Tests are downloaded using git
+    "-DWASMEDGE_BUILD_AOT_RUNTIME=OFF" # Fails otherwise
```

Comment:
Maybe it requires a newer llvm version or the clangStdenv?

Reply:
Just tried both but does not work. Their manual mentions llvm 12 and either of gcc or clang - https://wasmedge.org/book/en/extend/build.html

If you are able to figure it out, it would be helpful to know how you debugged it.

Reply:
Our llvm right now points to 11. clang and llvm are both in the same compiler famility. Try using clang12Stdenv instead of stdenv.

Reply:
As I said, I tried it already but fails with the same error. Not sure if it is relevant, but the AUR PKGBUILD also keeps that flag turned off.

Reply:
Can you update the comment with those information? I think a line or two should be enough.

Reply:
I just gave LLVM 11 a try and it worked! I did not think about it before since their manual has LLVM 12, so I tried 12,13, and 14.

Please squash the commit if everything looks fine.

Reply:
I was able to get this to build completely fine on LLVM 12, by using the LLVM and LLD packages from `llvmPackages_12` in `buildInputs`. Note that when you used `llvm_12` and `lld`, their versions were mismatched; maybe this was why? If upstream suggests using LLVM 12, we probably should as well.

Reply:
Thanks! That was it. I have pushed the changes now.